### PR TITLE
Clippy fixes

### PR DESCRIPTION
**Pull request by Caleb-Hill**

### CODE REVIEW

**Review by agunde406 — APPROVED** [review]

File: libsplinter/src/admin/rest_api/mod.rs [hunk 1]
```diff
@@ -90,13 +90,12 @@ impl RestResourceProvider for AdminService {
...
 #[derive(Clone)]
 pub struct CircuitResourceProvider {
-    node_id: String,
     store: Box<dyn AdminServiceStore>,
 }
 
 impl CircuitResourceProvider {
-    pub fn new(node_id: String, store: Box<dyn AdminServiceStore>) -> Self {
-        Self { node_id, store }
+    pub fn new(store: Box<dyn AdminServiceStore>) -> Self {
```
Comment (vaporos) [inline libsplinter/src/admin/rest_api/mod.rs:97]: Note this change is pub and should not be backported.

File: libsplinter/src/network/auth/mod.rs [hunk 2]
```diff
@@ -66,6 +66,7 @@ pub(crate) struct ManagedAuthorizationState {
     accepting_state: AuthorizationAcceptingState,
 
     // Tracks whether the local node has completed authorization with the remote node
+    #[cfg(any(feature = "trust-authorization", feature = "challenge-authorization"))]
```
Comment (vaporos) [inline libsplinter/src/network/auth/mod.rs:69]: Typo in commit - "feaures"

**Review by vaporos — CHANGES_REQUESTED** [review]
The commit which breaks compatibility with the current public API should explicitly state it breaks compatibility in the commit message.

File: libsplinter/src/admin/rest_api/mod.rs (hunk 1 above)
Comment (Caleb-Hill) [reply libsplinter/src/admin/rest_api/mod.rs:97]: Updated the commit message to say that

File: libsplinter/src/network/auth/mod.rs (hunk 2 above)
Comment (Caleb-Hill) [reply libsplinter/src/network/auth/mod.rs:69]: Fixed

**Review by agunde406 — APPROVED** [review]

**Review by vaporos — APPROVED** [review]

**Pull request merged**